src/organisms/Animal.js

```diff
@@ -0,0 +1,76 @@
+import { Organism } from '../Organism';
+import { getRandom } from '../utilities/getRandom';
+
+export class Animal extends Organism {
+  constructor(strength, initiative, positionY, positionX, age = 0) {
+    super(strength, initiative, positionY, positionX, age);
+  }
+
+  async action(board) {
+    const directions = [
+      { positionY: -1, positionX: -1 },
+      { positionY: 0, positionX: -1 },
+      { positionY: 1, positionX: -1 },
+      { positionY: -1, positionX: 0 },
+      { positionY: 1, positionX: 0 },
+      { positionY: -1, positionX: 1 },
+      { positionY: 0, positionX: 1 },
+      { positionY: 1, positionX: 1 },
+    ];
+
+    const availableDirections = directions.filter((direction) => {
+      const newY = this.positionY + direction.positionY;
+      const newX = this.positionX + direction.positionX;
+      return board.isValidPosition(newY, newX);
+    });
+
+    if (availableDirections.length > 0) {
+      const randomDirection =
+        availableDirections[getRandom(availableDirections.length)];
+      const newY = this.positionY + randomDirection.positionY;
+      const newX = this.positionX + randomDirection.positionX;
+
+      const targetOrganism = board.getOrganism(newY, newX);
+      if (!targetOrganism) {
+        board.moveOrganism(this, newY, newX);
+      } else if (targetOrganism.constructor.name === this.constructor.name) {
+        this.mate(board, targetOrganism);
+      } else {
+        this.fight(board, targetOrganism);
+      }
+    }
+  }
+
+  fight(board, opponent) {
+    if (opponent.constructor.name === 'Guarana') {
+      this.strength += 3;
+    }
```

Comment:
This logic shouldn't exist in the `Animal` class. It should be somewhere in the `Guarana` class. I see that you already have the `onEaten` method - use it.